mathics/builtin/strings.py

```diff
@@ -699,3 +699,182 @@ class StringQ(Test):
 
     def test(self, expr):
         return isinstance(expr, String)
+
+
+class StringTake(Builtin):
+    """
+    StringTake["string",n] gives the first n characters in "string"
+    StringTake["string",-n] gives the last n characters in "string"
+    StringTake["string",{m,n}] gives characters m through n in "string"
+    >> StringTake["abcde", 2]
+    = ab
+    >> StringTake["abcde", 0]
+    = 
+    >> StringTake["abcde", -2]
+    = de
+    >> StringTake["abcde", {2}]
+    = b
+    >> StringTake["abcd", {2,3}]
+    = bc
+    """
+    messages = {
+        'strse': 'String expected at position 1.',
+        'mseqs': 'Integer or list of two Intergers are expected \
+                  at position 2.',
+        'take': 'Cannot take positions `1` through `2` in "`3`\".',
+        }
+
+    def apply1_(self, string, n, evaluation):
+        'StringTake[string_,n_Integer]'
+        if not isinstance(string, String):
+            return evaluation.message('StringTake', 'strse')
+
+        pos = n.value
+        if pos > len(string.get_string_value()):
+            return evaluation.message('StringTake', 'take', 1, pos, string)
+        if pos < -len(string.get_string_value()):
+            return evaluation.message('StringTake', 'take', pos, -1, string)
+        if pos > 0:
+            return String(string.get_string_value()[:pos])
+        if pos < 0:
+            return String(string.get_string_value()[pos:])
+        if pos == 0:
+            return String("")             # it is what mma does
```

Comment:
Thats a test in the comments

    >> StringDrop["abcde", 2]
    = cde

And thats a hidden test:
    #> StringDrop["abcde", 2]
    = cde

I think (or #>>)